### PR TITLE
HPCC-29669 Better tracking of timing outliers when loading index nodes

### PR DESCRIPTION
**Pull request by ghalliday**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@mckellyln does this provide all the information we need?  Should I also add a stat reported by the query for time waiting for the critical section?  Is the is best target branch?

**Review by mckellyln — COMMENTED** [review]
Looks good, but did have a suggestion.

**Comment by ghalliday** [comment]
@mckellyln added the tracing you suggested and a stat for feeding back to the summary line

**Review by mckellyln — APPROVED** [review]
Yes, looks great.
Approved

File: system/jhtree/jhtree.cpp [hunk 1]
```diff
@@ -2614,10 +2638,38 @@ const CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offse
...
+                countExcessiveLock_x1++;
+
+            if ((endLoadCycles - lastLockingReportCycles) >= traceCacheLockingFrequency)
+            {
+                lastLockingReportCycles = endLoadCycles;
+                WARNLOG("CNodeCache::getNode lock(%s, %u) took %lluns  counts(>=%lluns, %u, %u, %u) (x1,x10,x100)", cacheTypeText[cacheType], whichCs, cycle_to_nanosec(lockingCycles),
+                         cycle_to_nanosec(traceCacheLockingThreshold),
+                         countExcessiveLock_x1.load() + countExcessiveLock_x10.load() + countExcessiveLock_x100.load(),
+                         countExcessiveLock_x10.load() + countExcessiveLock_x100.load(),
+                         countExcessiveLock_x100.load());
```
Comment (mckellyln) [inline system/jhtree/jhtree.cpp:2659]: Should we reset these here ?

File: system/jhtree/jhtree.cpp [hunk 2]
```diff
@@ -76,16 +76,18 @@ bool traceSmartStepping = false;
...
 MODULE_INIT(INIT_PRIORITY_JHTREE_JHTREE)
 {
     initCrit = new CriticalSection;
     fetchThresholdCycles = nanosec_to_cycle(defaultFetchThresholdNs);
-    traceCacheLockingFrequency = millisec_to_cycle(1000);           // Report locking delays at most once per second
+    traceCacheLockingFrequency = millisec_to_cycle(60000);          // Report locking delays at most once per minute
+    traceNodeLoadFrequency = millisec_to_cycle(60000);              // Report slow loads at most once per minute
     traceCacheLockingThreshold = millisec_to_cycle(50);             // Report locks that take > 50ms
-    traceNodeLoadThreshold = millisec_to_cycle(5);                  // Report node loads that take > 5ms
+    traceNodeLoadThreshold = millisec_to_cycle(20);                 // Report node loads that take > 5ms
```
Comment (mckellyln) [inline system/jhtree/jhtree.cpp:90]: comment - change to 20ms

File: system/jhtree/jhtree.cpp [hunk 3]
```diff
@@ -2662,6 +2662,9 @@ const CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offse
                          countExcessiveLock_x1.load() + countExcessiveLock_x10.load() + countExcessiveLock_x100.load(),
                          countExcessiveLock_x10.load() + countExcessiveLock_x100.load(),
                          countExcessiveLock_x100.load());
+                countExcessiveLock_x1 = 0;
+                countExcessiveLock_x10= 0;
```
Comment (mckellyln) [inline system/jhtree/jhtree.cpp:2666]: space before =

**Review by mckellyln — APPROVED** [review]
Approved.

File: system/jlib/jdebug.hpp [hunk 4]
```diff
@@ -30,6 +30,7 @@ __int64 jlib_decl cycle_to_nanosec(cycle_t cycles);
 __int64 jlib_decl cycle_to_microsec(cycle_t cycles);
 __int64 jlib_decl cycle_to_millisec(cycle_t cycles);
 cycle_t jlib_decl nanosec_to_cycle(__int64 cycles);
```
Comment (richardkchapman) [inline system/jlib/jdebug.hpp:32]: Not new but param name looks wrong

**Pull request merged**